src/env.cc

```diff
@@ -1157,7 +1164,7 @@ AsyncHooks::SerializeInfo AsyncHooks::Serialize(Local<Context> context,
   for (size_t i = 0; i < native_execution_async_resources_.size(); i++) {
     info.native_execution_async_resources[i] = creator->AddData(
         context,
-        native_execution_async_resources_[i].Get(context->GetIsolate()));
+        native_execution_async_resources_[i]);
```

Comment:
Shouldn't we assert that this the local is not empty before using it because we aren't taking ownership of it and it may get invalidated once the JS stack changes?

Reply:
Although the same as the comment above applies, thanks for bringing it up – `AddData()` has a `DCHECK` to verify that the object is not empty. I’ve made it so that we use `SIZE_MAX` as the index in the snapshot data struct to identify empty values (in a7787bbcabeeb8b63889c7baffdabb208ace999e).

Reply:
This too LGTM if the pointer inside the locals get nullified when they are GCed.